### PR TITLE
docs: add health checks to docs & examples

### PR DESCRIPTION
**Pull request by bpmct**
Added health checks to our documentation and example templates. Keeping as a draft until we release v0.9.1 which includes the improved health check behavior.

### CODE REVIEW

File: docs/ides/web-ides.md [hunk 1]
```diff
@@ -73,13 +79,16 @@ resource "coder_app" "code-server" {
...
+    interval  = 2
+    threshold = 10
+  }
+
 }
 ```
 
-<blockquote class="warning">
-If the code-server integrated terminal fails to load, (i.e., xterm fails to load), go to DevTools to ensure xterm is loaded, clear your browser cache and refresh.
-</blockquote>
```
Comment (bpmct) [inline docs/ides/web-ides.md:81]: Resolved by #3505

File: docs/ides/web-ides.md [hunk 2]
```diff
@@ -22,6 +22,12 @@ resource "coder_app" "portainer" {
   icon          = "https://simpleicons.org/icons/portainer.svg"
   url           = "http://localhost:8000"
   relative_path = true
+
+  healthcheck {
+    url       = "http://localhost:8000/"
```
Comment (f0ssel) [inline docs/ides/web-ides.md:27]: does portainer have a healthcheck route? If they have a helm chart you can see what they use for kubernetes.

File: dogfood/main.tf [hunk 3]
```diff
@@ -42,6 +42,12 @@ resource "coder_app" "code-server" {
   name     = "code-server"
   url      = "http://localhost:13337/"
   icon     = "/icon/code.svg"
+
+  healthcheck {
+    url       = "http://localhost:1337/healthz"
+    interval  = 3
```
Comment (f0ssel) [inline dogfood/main.tf:48]: I like lowering it to 3

**Review by f0ssel — APPROVED** [review]

File: docs/ides/web-ides.md (hunk 2 above)
Comment (bpmct) [reply docs/ides/web-ides.md:27]: Ah good call.

**Pull request merged**